include/zephyr/zbus/zbus.h

```diff
@@ -379,6 +389,37 @@ k_timeout_t _zbus_timeout_remainder(uint64_t end_ticks);
  */
 #define ZBUS_LISTENER_DEFINE(_name, _cb) ZBUS_LISTENER_DEFINE_WITH_ENABLE(_name, _cb, true)
 
+/**
+ * @brief Define and initialize a message subscriber.
+ *
+ * This macro defines an observer of message subscriber type. It defines a FIFO where the
+ * subscriber will receive the message asynchronously and initialize the @ref
+ * zbus_observer defining the subscriber.
+ *
+ * @param[in] _name The subscriber's name.
+ * @param[in] _enable The subscriber's initial state.
+ */
+#define ZBUS_MSG_SUBSCRIBER_DEFINE_WITH_ENABLE(_name, _enable)                                     \
+	static K_FIFO_DEFINE(_zbus_observer_fifo_##_name);                                         \
+	STRUCT_SECTION_ITERABLE(zbus_observer, _name) = {                                          \
+		ZBUS_OBSERVER_NAME_INIT(_name) /* Name field */                                    \
+			.type = ZBUS_OBSERVER_MSG_SUBSCRIBER_TYPE,                                 \
+		.enabled = _enable,                                                                \
+		.message_fifo = &_zbus_observer_fifo_##_name,                                      \
+	}
+
+/**
+ * @brief Define and initialize a message subscriber.
```

Comment:
The brief should be different as the other macro. 
But TBH I am not sure this additional macro is needed (duplicated doc, and little added value?), and I would just rename `ZBUS_MSG_SUBSCRIBER_DEFINE_WITH_ENABLE` as `ZBUS_MSG_SUBSCRIBER_DEFINE` as the "...WITH_ENABLE" naming feels a bit weird.

If you want to keep both macros then maybe:

```suggestion
 * @brief Define and initialize an enabled message subscriber.
```